packages/firestore/src/remote/remote_store.ts

```diff
@@ -315,16 +315,17 @@ export class RemoteStore {
     });
   }
 
-  private async onWatchStreamClose(
-    error: FirestoreError | null
-  ): Promise<void> {
+  private async onWatchStreamClose(error?: FirestoreError): Promise<void> {
     assert(
       this.isNetworkEnabled(),
       'onWatchStreamClose() should only be called when the network is enabled'
     );
 
     this.cleanUpWatchStreamState();
-    this.onlineStateTracker.handleWatchStreamFailure();
+
+    if (error) {
```

Comment:
I am tempted to make this part of the `this.shouldStartWatchStream()` if-check. That way we only change the state to `Unknown` if we actually care (and nicely works around the problem that the comment in line 330 seems to apply more to line 326 than 331).

Reply:
done (and tweaked comments)